halo2_proofs/src/multicore.rs

```diff
@@ -2,8 +2,15 @@
 //! `halo2`. It's currently just a (very!) thin wrapper around [`rayon`] but may
 //! be extended in the future to allow for various parallelism strategies.
 
-pub use rayon::{
-    current_num_threads,
+pub use maybe_rayon::{
     iter::{IndexedParallelIterator, IntoParallelIterator, ParallelIterator},
```

Comment:
Please investigate the wasm32-wasi failure; I believe this was working on that platform before.